src/classes/BDI_DataImportService.cls

```diff
@@ -878,6 +878,18 @@ global with sharing class BDI_DataImportService {
                 return false;
             }
 
+            // If the destination field is a Boolean and the DI field intermediate requires interpretation to Boolean
+            // This allows for mapping checkboxes to a tri-state picklist - None, True, False - or a text with either 
+            // blank, "True", or "False"
+            String destinationSObjectname = destinationRecord.getSObjectType().getDescribe().getName(); 
+            if (!(value instanceof Boolean) && 
+                (UTIL_Describe.getFieldDisplayType(UTIL_Namespace.StrTokenNSPrefix(destinationSObjectname), 
+                 UTIL_Namespace.StrTokenNSPrefix(destinationField.toLowerCase())) == 
+                Schema.DisplayType.BOOLEAN)) {
+
+                value = Boolean.valueOf(value);
```

Comment:
While you are inside this method, it might be nice to rename the `value` var here to something like `sourceField` but just my preference 😄 .

Reply:
nm - I realize why it is named this way now!